### ISSUE TITLE
CA1861 makes a breaking suggestion for attribute arguments

### ISSUE THREAD

**Issue by Kaoticz** [dotnet/roslyn-analyzers#7205]
### Analyzer

**Diagnostic ID**: [CA1861](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1861): `Avoid constant arrays as arguments`

### Analyzer source

**Version**: [SDK 8.0.101](https://dotnet.microsoft.com/download/dotnet/8.0)

### Describe the bug

CA1861 triggers when an array is initialized as an argument to an attribute. This shouldn't happen, as attributes **require** constant arguments. Therefore, the CA1861 suggestion of initializing a static array is a breaking suggestion, leading to code that does not compile.

Found it while writing an **xUnit** test:

```cs
[Theory]
[InlineData(new int[] { })]	// CA1861 triggers here. It should not.
[InlineData(new int[] { 0, 1, 2, 3 })] // CA1861 triggers here. It should not.
internal void SomeTest(IReadOnlyList<int> collection)
{
    // test code...
}
```

If you apply the suggested correction, you get [CS0182](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/parameter-argument-mismatch?f1url=%3FappId%3Droslyn%26k%3Dk%28CS0182%29#arguments-for-attributes) (*An attribute argument must be a constant expression, `typeof` expression or array creation expression of an attribute parameter type*), which is a compilation error.

This issue seems to be related to https://github.com/dotnet/roslyn-analyzers/issues/7033

### Steps To Reproduce

Initialize an array for an attribute argument.
```cs
[SomeAttribute(new object[] { })]
```

### Expected behavior

No suggestion, warning, or error should be made.

### Actual behavior

A CA1861 suggestion is made.

## Additional context

VS Code + C# Dev Kit 1.3.10
.NET 8.0.101

**Comment by CollinAlpert** [comment]
Since this issue is already fixed, I suggest you install the `Microsoft.CodeAnalysis.NetAnalyzers` package with version `9.0.0-preview.24072.1`

**Comment by cw397** [comment]
I can confirm it is fixed for us in that preview package. From the dates it looks like the fix just missed the cut for version 8. Does this mean the fix won't be in the SDK until .NET 9? Is there any likelihood of a patch release for version 8? There were several for version 7.

**Comment by CollinAlpert** [comment]
I'm not sure if analyzers are included in the patch releases of an SDK. Someone from the team would need to comment on that.

**Comment by cw397** [comment]
Even a patch release just of the analyzers might be helpful. It was that I was thinking of more when I said about the ones for version 7.

**Comment by CollinAlpert** [comment]
This NuGet package contains the fix: https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers/9.0.0-preview.24072.1

**Comment by Morilli** [comment]
I just ran into this same issue. Using the nuget package `Microsoft.CodeAnalysis.NetAnalyzers` 9.0.0 fixes it, but building with the .NET 9 SDK without that package still shows the CA1861 warning, even though the .NET 9 SDK has a newer version, so something is still wrong.

**Comment by steveberdy** [comment]
@Morilli which precise version 9.0 did you use before .NET 9? A false-positive fix may have come after the version shipped with .NET 9.

**Comment by Morilli** [comment]
Testing this again, I cannot reproduce this inspection again with any version of the .NET 9 SDK, just the .NET 8 one. So it looks like some inspection results may have been cached incorrectly or I had forgotten to switch back to the .NET 9 SDK while testing.

All versions of the .NET 9 SDK look fine for me now.
